### PR TITLE
PCE exclude subgraph design

### PR DESCRIPTION
**Pull request by IlyaMoiseev**
Design for #1451

### CODE REVIEW

File: docs/design/solutions/pce-exclude-subgraph/pce-exclude-subgraph.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+## Goals
+Add general excluding mechanics in Computation Engine. It will be used in scenarios like:
+- Calculate a diverse path
+- Evacuate flows from ISL/Switch (reroute with exclude)
+- Allow user to specify exclusion list from API
+
+## Solution overview
+There are two rules of excluding
+- Exclude specified ISL
+- Exclude entire switch
```
Comment (surabujin) [inline docs/design/solutions/pce-exclude-subgraph/pce-exclude-subgraph.md]: Excluding switch you exclude all ISL attached to this switch. So switch exclusion is equal to ISL exclusion.

In other word we need to implement exclusion of N ISL from path computation.

**Comment by IlyaMoiseev** [comment]
@surabujin I'm looking into current PCE with DAO branch, AvailableNetwork instance became internal there: https://github.com/telstra/open-kilda/blob/de9089f14c63db3d579b7efd3d77957ae4db3e05/services/src/kilda-pce/src/main/java/org/openkilda/pce/impl/PathComputerImpl.java#L69
Maybe it requires some refactoring for this feature

File: docs/design/solutions/pce-exclude-subgraph/pce-exclude-subgraph.md (hunk 1 above)
Comment (pzakatov) [reply docs/design/solutions/pce-exclude-subgraph/pce-exclude-subgraph.md]: I'd vote for keeping switch filter and do a performance test afterward, so we can decide later. Having this filtering logic inside PCE will simplify external business logic a bit.

Comment (surabujin) [reply docs/design/solutions/pce-exclude-subgraph/pce-exclude-subgraph.md]: We can and should have "exclude switch" into NB API.

The possibility to convert "exclude switch" into "exclude N ISL" is our implementation details and external world should not worry/know about them.

**Comment by IlyaMoiseev** [comment]
Diagrams updated

**Pull request merged**